File: example/python/line_styles/line_styles.py
#!/usr/bin/env python3
"""
Line styles demonstration - Dual mode: fortplotlib or matplotlib
Equivalent to line_styles.f90 for visual comparison
"""

import sys
import numpy as np

# Dual-mode import: --matplotlib uses matplotlib, default uses fortplotlib
if "--matplotlib" in sys.argv:
    import matplotlib.pyplot as plt
    backend = "matplotlib"
else:
    import fortplotlib.fortplot as plt
    backend = "fortplotlib"

def main():
    print(f"=== Line Style Examples ({backend}) ===")
    
    # Generate test data with clear separation for visibility (same as Fortran)
    x = np.arange(50) * 0.2
    y1 = np.sin(x) + 2.0
    y2 = np.cos(x) + 1.0
    y3 = np.sin(x * 2.0)
    y4 = np.cos(x * 3.0) - 1.0
    y5 = np.sin(x * 0.5) - 2.0
    
    # Comprehensive line style demonstration
    plt.figure(figsize=(8, 6))
    plt.plot(x, y1, label='Solid (-)', linestyle='-')
    plt.plot(x, y2, label='Dashed (--)', linestyle='--')
    plt.plot(x, y3, label='Dotted (:)', linestyle=':')
    plt.plot(x, y4, label='Dash-dot (-.)', linestyle='-.')
    
    plt.title('Complete Line Style Reference')
    plt.xlabel('X values')
    plt.ylabel('Y values')
    plt.legend()
    
    plt.savefig('line_styles.png')
    plt.savefig('line_styles.pdf')
    
    # Save TXT for fortplotlib only
    if backend == "fortplotlib":
        plt.savefig('line_styles.txt')
    
    if backend == "matplotlib":
        plt.close()
    
    print("Line style examples completed!")
    print("Files created: line_styles.png/pdf" + ("" if backend == "matplotlib" else "/txt"))

if __name__ == "__main__":
    main()

File: example/fortran/line_styles/line_styles.f90
program line_styles
    !! Comprehensive demonstration of line styles in fortplotlib
    !!
    !! This example shows all available line styles:
    !! - Solid lines (-)
    !! - Dashed lines (--)
    !! - Dotted lines (:)
    !! - Dash-dot lines (-.)
    !! - No line (None)
    use fortplot
    implicit none

    real(wp), dimension(50) :: x, y1, y2, y3, y4, y5, y6
    integer :: i

    print *, "=== Line Style Examples ==="

    ! Generate test data with clear separation for visibility
    do i = 1, 50
        x(i) = real(i-1, wp) * 0.2_wp
        y1(i) = sin(x(i)) + 2.5_wp
        y2(i) = cos(x(i)) + 1.5_wp
        y3(i) = sin(x(i) * 2.0_wp) + 0.5_wp
        y4(i) = cos(x(i) * 3.0_wp) - 0.5_wp
        y5(i) = sin(x(i) * 0.5_wp) - 1.5_wp
        y6(i) = cos(x(i) * 0.5_wp) - 2.5_wp
    end do

    ! Comprehensive line style demonstration
    call figure(800, 600)
    call plot(x, y1, label='Solid (-)', linestyle=LINESTYLE_SOLID)
    call plot(x, y2, label='Dashed (--)', linestyle=LINESTYLE_DASHED)
    call plot(x, y3, label='Dotted (:)', linestyle=LINESTYLE_DOTTED)
    call plot(x, y4, label='Dash-dot (-.)', linestyle=LINESTYLE_DASHDOT)
    call plot(x, y5, label='None (invisible)', linestyle=LINESTYLE_NONE)
    call plot(x, y6, label='Markers only', linestyle='o')
    call title('Complete Line Style Reference')
    call xlabel('X values')
    call ylabel('Y values')
    call legend()
    call savefig('output/example/fortran/line_styles/line_styles.png')
    call savefig('output/example/fortran/line_styles/line_styles.pdf')
    call savefig('output/example/fortran/line_styles/line_styles.txt')

    print *, "Line style examples completed!"
    print *, "Files created: line_style_reference.png, line_style_reference.pdf"
    print *, "Note: 'None' linestyle creates invisible lines (no output)"
    print *

end program line_styles
